src/coreclr/src/vm/debuginfostore.cpp

```diff
@@ -552,6 +574,20 @@ void CompressDebugInfo::RestoreBoundariesAndVars(
     if (pcVars != NULL) *pcVars = 0;
     if (ppVars != NULL) *ppVars = NULL;
 
+    // Check flag byte and skip over any patchpoint info
```

Comment:
This is changing the debug info format. The debug info format is part of R2R format, so changing it would need to touch number of places (e.g. src\coreclr\src\tools\crossgen2\ILCompiler.Reflection.ReadyToRun\DebugInfo.cs).

It may be easiest to append the extra stuff at the end, and look for it only when the method has patch info (e.g. only for JITed methods).

Reply:
Sure, let me rework this.

Reply:
Have looked at adding patchpoint info to the end of the compressed debug info blob, with the idea that some of the existing debug info producers/consumers would not need to change, but this seems fragile.

The current header record for the debug info just has the lengths of the two encoded sub-records. So there's no way to tell if what lies beyond the sub records is really patchpoint info or just random bytes.

Modifying the header to include more length info is similar in spirit to what I'm already doing. Adding a new header field -- either for overall length, for header length, or for patchpoint info length -- will require all producers and consumers to change.

Reply:
Updated so only jitted code debug is impacted, and only when feature is defined.

Still may want to tweak this, say by putting the patchpoint info length in the header instead of a flag byte.